### PR TITLE
Dockerized version of workspace

### PR DESCRIPTION
**Pull request by sychan**
The main codebase is unaffected, just a lot of files to support building a docker image, and bringing up a viable workspace using docker-compose.

It would be good to modify the workspace tests so that they can run against the dockerized service. I'm wondering if this would be a useful task for someone like Josh so that he gains some familiarity with the workspace?

### CODE REVIEW

**Comment by MrCreosote** [comment]
> It would be good to modify the workspace tests so that they can run against the dockerized service. I'm wondering if this would be a useful task for someone like Josh so that he gains some familiarity with the workspace?

From our discussion, it'd make a lot more sense just to write the small integration test you wanted to ensure workspace is talking to Shock and the HS/M correctly. Running the full test suite is massive overkill.

**Comment by MrCreosote** [comment]
Per our discussion @sychan is going to move as much stuff as possible into a miniKBase repo (or something) and leave the minimum amount of stuff to create the WS docker container

File: server_scripts/glassfish_administer_service.py [hunk 1]
```diff
@@ -297,8 +296,9 @@ def _run_remote_command(self, *cmd):
 
 if __name__ == '__main__':
     args = _parseArgs()
+    _VERBOSE = args.verbose
```
Comment (MrCreosote) [inline server_scripts/glassfish_administer_service.py]: Don't make this a global variable, you philistine. Pass it into the CommandGlassfishDomain class and make it an instance variable.

**Comment by sychan** [comment]
I'm going to move a bunch of stuff into a new repo and submit a new PR

**Pull request closed without merging**